packages/vats/src/virtual-purse.js

```diff
@@ -143,6 +145,18 @@ function makeVirtualPurse(vpc, kit) {
       });
       return pmt;
     },
+    claim: (_paymentP, _optAmountShape) => {
+      // TODO implement
+      assert.fail(X`virtual purses do not yet implement claim`);
+    },
+    getRecoverySet: () => {
```

Comment:
This should be implemented. It looks to me like it's exactly the same code; all the recovery is on the JS side.

Reply:
What I'm confused about here is whether a vpurse
   * a vpurse should forward to an underlying ERTP purse which implements this logic, or 
   * should reimplement that on its own.
   * one case of vpurse should forward and the other case should reimplement.

I suspect the third bullet is the right answer, but I'm not well enough oriented to be confident.

@michaelfig can you do the vpurse adjustment and I'll review?

@dtribble if yes, would you rather @michaelfig do that in a separate PR or as an enhancement to this PR?

Either way, when this PR is ready, someone else should merge it if I don't seem to be available.

Reply:
Thanks

Reply:
> - should reimplement that on its own

In our uses of this `virtual-purse.js`, each vpurse should implement its own recoverySet, so that recoverAll calls vpurse.deposit (which is where the vpurse magic happens).  I think the right thing is to make it part of the current PR.  I can attempt this if you don't have bandwidth for now.

Note that vpurses are not yet Virtual nor Durable objects, and even without that restriction, are somewhat simpler than `@agoric/ertp` purses.  That makes things easier.

Reply:
>  I can attempt this if you don't have bandwidth for now.

Yes please! Much appreciated!